### DISCUSSION TITLE
Link causes refresh in my app

### DISCUSSION THREAD

**Discussion by MehediH** [vercel/next.js#13822] [Help] [answered]
Hi, I am trying out Incremental Static Regeneration for the first time, and it's super cool! However, I am running into an issue where the incrementally generated route causes a refresh.

To be more specific, I have a page with a couple of `<Link>` elements that link to these incrementally generated routes. I don't generate these routes at build (because of API rate limiting and the fact that these routes may not be opened much by the user). 

When I click on these links, it navigates to the newly generated page, except it does not retain the state of the page, as it causes a refresh. Is there any way I can make the navigations to the incrementally generated routes act like the "usual" navigation between pages within the app?

Thanks!

**Comment by Timer** [comment, marked as answer]
It sounds like you're improperly using `<Link>`. Could you post your links?

**Reply by MehediH** [reply]
Yep, sure. Here's the code for the `<Link>`

```
<Link href={`/apps/${app._id}`}>
    <a>View App</a>
</Link>
```

For the incrementally generated page, here's my code:

```
export async function getStaticPaths() {
    return {
      paths: [],
      fallback: true,
    };
}

export async function getStaticProps({ params }) {
    let app = await fetch(`example.api`).then((res) =>
        res.json()
    );

    return {
        props: {
            app
        },
        unstable_revalidate: 1
    }
}
```

Let me know if that helps or you need any additional details, thank you!

**Reply by Timer** [reply]
That's all I needed to see! Your `<Link>` is incorrect.

Please see the documentation:
https://nextjs.org/docs/api-reference/next/link#dynamic-routes

What is your page file name?

**Reply by MehediH** [reply]
Aha! That fixed it. Thank you so much!
